### PR TITLE
Python testing style guide

### LINKED ISSUE #7: Testing workflows
Add information about testing workflows.

### PR DESCRIPTION
**Pull request by pllim**
Fix #7 

p.s. I hope I didn't derail too much. @drdavella and @SaOgaz , please have a look.

### CODE REVIEW

File: guides/python-testing.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+## Where should I put my tests?
+
+See the templates in Python Packaging section.
+
+Template aside, there is also the question of whether you want to bundle the
+tests with your package or not. While bundling allows users to test their
+installations, it also increases the package size. For more information,
+see "Choosing a test layout / import rules" in
+https://docs.pytest.org/en/latest/goodpractices.html .
```
Comment (drdavella) [inline guides/python-testing.md]: There are maybe two separate decisions to be made here: 
1. Should the tests live in the source tree or outside of it entirely?
2. If the tests live outside of the source tree, should they still be included in the package distribution?

Comment (pllim) [reply guides/python-testing.md]: Do you have a short text suggestion that would address these questions? I do not want this document to be a lengthy philosophical essay. Just short and sweet for people to get started with. And if they are interested to learn more, there is always the Internet.

Comment (pllim) [reply guides/python-testing.md]: Feel free to edit this, Dan. You are a co-author. :wink:  :pray:  Maybe I missed something. Do we need to talk about `tox`?

File: guides/python-testing.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+## What should I test?
+
+In theory, there are many kinds of software testing. For instance,
+unit testing, integration testing, regression testing, etc. Currently, this
+document only focuses on unit and regression tests, which are usually
```
Comment (SaOgaz) [inline guides/python-testing.md]: You note here that you will only cover unit and regression tests, but then there's a bit at the end talking about CI testing....

Comment (pllim) [reply guides/python-testing.md]: I am still not sure where is the boundary between regression and CI. In my experience, they both go hand in hand. Please suggest better verbiage if possible. 😅

Comment (SaOgaz) [reply guides/python-testing.md]: Maybe something like this...

"Currently, this document focuses on unit and regression tests (usually "shipped" with the package they are tied to), as well as continuous integration testing.

File: guides/python-testing.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+versions and operating systems when a pull request is opened or modified).
+On the other hand, you might also want to consider only running certain
+unit tests in regression mode, especially ones that run slow or generate
+a lot of network hits, which might exceed time out quota of certain services.
+
+### Tests doomed to fail
+
+See https://docs.pytest.org/en/latest/skipping.html .
+
+### Testing with outside dependancies
```
Comment (pllim) [inline guides/python-testing.md]:
```suggestion
### Testing with outside dependencies
```

**Comment by pllim** [comment]
Thanks, @stscicrawford ! LGTM.

**Pull request merged**